### PR TITLE
docker: Add iopath channel, have docker build be testable

### LINKED ISSUE #63342: Nightly Docker image builds are failing with conda install issues (build-publish-docker)
See: https://github.com/pytorch/pytorch/runs/3340344859

Appears to be failing on the install step for nightlies:

https://github.com/pytorch/pytorch/blob/979180cd0118f9379680df9e39fc6acb80d7b80a/Dockerfile#L58-L59

I think this is related to torchtext since I only get installation issues with the following command when I add torchtext:

```
nerdctl run --rm -it continuumio/miniconda3 sh -c 'conda install -c "pytorch-nightly" -c "nvidia" -y python=3.7 pytorch torchvision torchaudio torchtext "cudatoolkit=11.1"'
```

cc @ezyang @gchanan @zou3519 @bdhirsh @jbschlosser @anjali411 @malfet @seemethere @walterddr

### PR DESCRIPTION
**Pull request by seemethere**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #63811

Closes https://github.com/pytorch/pytorch/issues/63342

Temporary fix for https://github.com/pytorch/text/issues/1380

Signed-off-by: Eli Uriegas <eliuriegas@fb.com>

Differential Revision: [D30497559](https://our.internmc.facebook.com/intern/diff/D30497559)

cc @ezyang @seemethere @malfet @walterddr

### CODE REVIEW

**Comment by seemethere** [comment]
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30497559).

**Comment by seemethere** [comment]
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30497559).

**Review by malfet — APPROVED** [review]

**Comment by seemethere** [comment]
@seemethere has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30497559).

File: docker.Makefile [hunk 1]
```diff
@@ -22,7 +22,7 @@ PYTHON_VERSION            = 3.7
 PYTORCH_VERSION           = $(shell git describe --tags --always)
 # Can be either official / dev
 BUILD_TYPE                = dev
-BUILD_PROGRESS            = auto
+BUILD_PROGRESS           ?= auto
```
Comment (seemethere) [inline docker.Makefile:25]: Changed so we can pass BUILD_PROGRESS as an env variable from the actions workflow

**Comment by seemethere** [comment]
Closing since https://github.com/pytorch/text/pull/1381 has been merged

**Pull request closed without merging**